src/test_JS_repo_lib.py

```diff
@@ -1,471 +1,551 @@
-		if retcode != 0 and build_scripts.count(b) < 2:
-			build_debug += "ERROR running command: " + b
-			build_scripts += [b] # re-add it onto the end of the list, and try running it again after the other build commands
-		elif retcode == 0:
-			build_script_list += [b]
-	return( retcode, build_script_list, build_debug)
-
-def run_tests( manager, pkg_json, crawler):
-	test_json_summary = {}
-	retcode = 0
-	if len(crawler.TRACKED_TEST_COMMANDS) == 0:
-		return(retcode, test_json_summary)
-	test_scripts = [t for t in pkg_json.get("scripts", {}).keys() if not set([ t.find(t_com) for t_com in crawler.TRACKED_TEST_COMMANDS]) == {-1}]
-	test_scripts = [t for t in test_scripts if set([t.find(ig_com) for ig_com in crawler.IGNORED_COMMANDS]) == {-1}]
-	test_scripts = [t for t in test_scripts if set([pkg_json.get("scripts", {})[t].find(ig_sub) for ig_sub in crawler.IGNORED_SUBSTRINGS]) == {-1}]
-	for t in test_scripts:
-		print("Running: " + manager + t)
-		error, output, retcode = run_command( manager + t, crawler.TEST_TIMEOUT)
-		test_info = TestInfo( (retcode == 0), error, output, manager, crawler.VERBOSE_MODE)
-		test_info.set_test_command( pkg_json.get("scripts", {})[t])
-		test_info.compute_test_infras()
-		test_info.compute_nested_test_commands( test_scripts)
-		test_info.compute_test_stats()
-		# print( test_info[t])
-		# print( get_test_info(error, output))
-		test_json_summary[t] = test_info.get_json_rep()
-	return( retcode, test_json_summary)
-
-def called_in_command( str_comm, command, manager):
-	# command ends with command terminator (this list includes \0 end-of-string, 
-	# but this is not available to check in Python so we use endswith)
-	post_command_chars = [ "" ] if command.endswith(str_comm) else [ " ", "\t", ";"]
-	for pcc in post_command_chars:
-		check_comm = str_comm + pcc
-		if command.find( check_comm) == 0:
-			return( True)
-		if command.find( "&&" + check_comm) > -1 or command.find( "&& " + check_comm) > -1:
-			return( True)
-		if command.find( "cross-env NODE_ENV=test " + check_comm) > -1 or command.find( "cross-env NODE_ENV=production " + check_comm) > -1:
-			return( True)
-		if command.find( "cross-env CI=true " + check_comm) > -1:
-			return( True)
-		if command.find( "opener " + check_comm) > -1:
-			return( True)
-		if command.find( "gulp " + check_comm) > -1:
-			return( True)
-		if command.find( "nyc " + check_comm) > -1:
-			return( True)
-	return( False)
-
-def test_cond_count( test_output, regex_fct, condition, offset):
-	ptrn = re.compile( regex_fct(condition), re.MULTILINE)
-	results = ptrn.findall( test_output)
-	if offset is None:
-		return( len( results)) # just count the number of hits, each hit is an individual test (example: tap "ok" vs "not ok")
-	num_cond = 0
-	for r in results:
-		temp = r.split()
-		try:
-			num_cond += int( temp[temp.index(condition) + offset])  
-		except ValueError:
-			num_cond += 0
-	return( num_cond)
+
+
+def get_fixDeps():
```

Comment:
please rename to snake case to match the rest of the code; maybe just call it `get_all_deps`